kotlin-css/src/main/kotlin/kotlinx/css/StyleEnums.kt

```diff
@@ -89,237 +95,418 @@ enum class Clear {
+        val paleTurquoise = Color("paleturquoise", "#afeeee")
+        val paleVioletRed = Color("palevioletred", "#db7093")
+        val papayaWhip = Color("papayawhip", "#ffefd5")
+        val peachPuff = Color("peachpuff", "#ffdab9")
+        val peru = Color("peru", "#cd853f")
+        val pink = Color("pink", "#ffc0cb")
+        val plum = Color("plum", "#dda0dd")
+        val powderBlue = Color("powderblue", "#b0e0e6")
+        val purple = Color("purple", "#800080")
+        val red = Color("red", "#ff0000")
+        val rosyBrown = Color("rosybrown", "#bc8f8f")
+        val royalBlue = Color("royalblue", "#4169e1")
+        val saddleBrown = Color("saddlebrown", "#8b4513")
+        val salmon = Color("salmon", "#fa8072")
+        val sandyBrown = Color("sandybrown", "#f4a460")
+        val seaGreen = Color("seagreen", "#2e8b57")
+        val seaShell = Color("seashell", "#fff5ee")
+        val sienna = Color("sienna", "#a0522d")
+        val silver = Color("silver", "#c0c0c0")
+        val skyBlue = Color("skyblue", "#87ceeb")
+        val slateBlue = Color("slateblue", "#6a5acd")
+        val slateGray = Color("slategray", "#708090")
+        val snow = Color("snow", "#fffafa")
+        val springGreen = Color("springgreen", "#00ff7f")
+        val steelBlue = Color("steelblue", "#4682b4")
+        val tan = Color("tan", "#d2b48c")
+        val teal = Color("teal", "#008080")
+        val thistle = Color("thistle", "#d8bfd8")
+        val tomato = Color("tomato", "#ff6347")
+        val turquoise = Color("turquoise", "#40e0d0")
+        val violet = Color("violet", "#ee82ee")
+        val wheat = Color("wheat", "#f5deb3")
+        val white = Color("white", "#ffffff")
+        val whiteSmoke = Color("whitesmoke", "#f5f5f5")
+        val yellow = Color("yellow", "#ffff00")
+        val yellowGreen = Color("yellowgreen", "#9acd3")
+
+        val normalizeFractionalPercent = { v: Double -> normalize<Double>(min = 0.0, max = 1.0, value = v) }
+        val normalizePercent = { v: Int -> normalize<Int>(min = 0, max = 100, value = v) }
+        val normalizeRGB = { v: Int -> normalize<Int>(min = 0, max = 255, value = v) }
+        val normalizeAlpha = normalizeFractionalPercent
+        val normalizeHue: (Double) -> Int = { v: Double -> (((v % 360) + 360) % 360).roundToInt() }   // algorithm for capping from W3C
     }
 
     override fun toString() = value
 
-    fun withAlpha(alpha: Double): Color {
-        if (alpha < 0 || alpha > 1) {
-            throw IllegalArgumentException("Alpha should be a number between 0.0 (fully transparent) and 1.0 (fully opaque)")
-        } else {
-            val rgba = toRGBA()
-            return Color("rgba(${rgba.red},${rgba.green},${rgba.blue},${alpha * rgba.alpha})")
+    fun withAlpha(alpha: Double) =
+        when {
+            value.startsWith("hsl", true) -> with(fromHSLANotation()) { hsla(hue, saturation, lightness, normalizeAlpha(alpha) * this.alpha) }
```

Comment:
I don't get the `normalizeAlpha(alpha) * this.alpha` bit, I think it should be just `normalizeAlpha(alpha)`

Reply:
The [original code](https://github.com/JetBrains/kotlin-wrappers/blob/50336a4fe3ad3251716616525071028cb1a635cf/kotlin-css/src/main/kotlin/kotlinx/css/StyleEnums.kt#L258) was:
```kotlin
return Color("rgba(${rgba.red},${rgba.green},${rgba.blue},${alpha * rgba.alpha})")
```
Bascially, multiplying the *new alpha* with the *original alpha*.  The ``normalizeAlpha()`` call just ensures that the new alpha is in the range of 0.0 to 1.0.

Reply:
Sorry, I get it now. I've added more tests and they all pass. Still, when I integrate the new build into my project, `blackAlpha` stops working correctly.

Reply:
This one proved to be quite tricky. The `normalize` function that was written using generic types was not working correctly in Kotlin/JS, after several attempts at fixing it I rewrote it without generics.

Reply:
@Leonya Is that an issue that should be reported?

Reply:
Yes, I'll talk to the Kotlin team on Monday.